log4j-core-test/src/test/java/org/apache/logging/log4j/core/appender/rolling/PatternProcessorTest.java

```diff
@@ -274,4 +310,171 @@ public void testGetNextTimeWeeklyReturnsFirstWeekInYear_US() {
+
+        // expect Wed, March 5, 2014, 00:00
+        final Calendar expected = Calendar.getInstance(TimeZone.getTimeZone("GMT-6"));
+        expected.set(2014, Calendar.MARCH, 5, 0, 0, 0);
+        expected.set(Calendar.MILLISECOND, 0);
+        assertEquals(format(expected.getTimeInMillis()), format(actual));
+    }
+
+    @Test
+    @ResourceLock(value = Resources.LOCALE, mode = ResourceAccessMode.READ)
+    @ResourceLock(value = Resources.TIME_ZONE, mode = ResourceAccessMode.READ)
+    public void testGetNextTimeDailyReturnsFirstHourOfNextDayHonoringTimeZoneOption2() {
+        TimeZone old = TimeZone.getDefault();
+        TimeZone.setDefault(TimeZone.getTimeZone("GMT+10")); // default is ignored if pattern contains timezone
+        try {
+            final PatternProcessor pp = new PatternProcessor("logs/app-%d{yyyy-MM-dd}{GMT-6}.log.gz");
+            final Calendar initial = Calendar.getInstance(TimeZone.getTimeZone("GMT-6"));
+            initial.set(2014, Calendar.MARCH, 4, 2, 31, 59); // Tue, March 4, 2014, 02:31
+            final long actual = pp.getNextTime(initial.getTimeInMillis(), 1, false);
+
+            // expect Wed, March 5, 2014, 00:00
+            final Calendar expected = Calendar.getInstance(TimeZone.getTimeZone("GMT-6"));
+            expected.set(2014, Calendar.MARCH, 5, 0, 0, 0);
+            expected.set(Calendar.MILLISECOND, 0);
+            assertEquals(format(expected.getTimeInMillis()), format(actual));
+        } finally {
+            TimeZone.setDefault(old);
+        }
+    }
+
+    @Test
+    @ResourceLock(value = Resources.LOCALE, mode = ResourceAccessMode.READ)
+    @ResourceLock(value = Resources.TIME_ZONE, mode = ResourceAccessMode.READ)
+    public void testGetNextTimeDailyReturnsFirstHourOfNextDayHonoringTimeZoneOption3() {
+        TimeZone old = TimeZone.getDefault();
+        TimeZone.setDefault(TimeZone.getTimeZone("GMT-10")); // default is ignored if pattern contains timezone
+        try {
+            final PatternProcessor pp = new PatternProcessor("logs/app-%d{yyyy-MM-dd}{GMT-6}.log.gz");
+            final Calendar initial = Calendar.getInstance(TimeZone.getTimeZone("GMT-6"));
+            initial.set(2014, Calendar.MARCH, 4, 2, 31, 59); // Tue, March 4, 2014, 02:31
+            final long actual = pp.getNextTime(initial.getTimeInMillis(), 1, false);
+
+            // expect Wed, March 5, 2014, 00:00
+            final Calendar expected = Calendar.getInstance(TimeZone.getTimeZone("GMT-6"));
+            expected.set(2014, Calendar.MARCH, 5, 0, 0, 0);
+            expected.set(Calendar.MILLISECOND, 0);
+            assertEquals(format(expected.getTimeInMillis()), format(actual));
+        } finally {
+            TimeZone.setDefault(old);
+        }
+    }
+
+    @Test
+    @ResourceLock(value = Resources.LOCALE, mode = ResourceAccessMode.READ)
+    @ResourceLock(value = Resources.TIME_ZONE, mode = ResourceAccessMode.READ)
```

Comment:
In this test and the remaining ones I don't think we need to lock `Resources.TIME_ZONE`: the result of the test should **not** depend on the default timezone.